contentcuration/contentcuration/db/models/manager.py

```diff
@@ -74,54 +122,55 @@ def _move_child_to_new_tree(self, node, target, position):
             Q(prerequisite_id=node.id) | Q(target_node_id=node.id)
         ).delete()
 
-    @contextlib.contextmanager
-    def _update_changes(self, node, save):
-        original_parent_id = node.parent_id
-        yield
-        ids = [original_parent_id, node.parent_id] + [node.id] if save else []
-        # Always write to the database for the parent change updates, as we have
-        # no persistent object references for the original and new parent to modify
-        if ids:
-            self.filter(id__in=ids).update(changed=True)
-        node.changed = True
-
-    def _move_node(
-        self, node, target, position="last-child", save=True, refresh_target=True
-    ):
-        # If we are delaying updates, then _move_node defers to insert_node
-        # we are already wrapping for parent changes below, so no need to
-        # add our parent changes context manager.
-        if self.tree_model._mptt_is_tracking:
-            return super(CustomContentNodeTreeManager, self)._move_node(
-                node, target, position, save, refresh_target
-            )
-        with self._update_changes(node, save):
-            return super(CustomContentNodeTreeManager, self)._move_node(
-                node, target, position, save, refresh_target
-            )
-
-    def insert_node(
-        self,
-        node,
-        target,
-        position="last-child",
-        save=False,
-        allow_existing_pk=False,
-        refresh_target=True,
-    ):
-        with self._update_changes(node, save):
-            if save:
-                with self.lock_mptt(target.tree_id):
-                    return super(CustomContentNodeTreeManager, self).insert_node(
-                        node, target, position, save, allow_existing_pk, refresh_target
-                    )
-            return super(CustomContentNodeTreeManager, self).insert_node(
-                node, target, position, save, allow_existing_pk, refresh_target
+    def _mptt_refresh(self, *nodes):
+        ids = [node.id for node in nodes if node.id]
+        if not ids:
+            return
+        opts = self.model._mptt_meta
+        values_lookup = {
+            c["id"]: c
+            for c in self.filter(id__in=ids).values(
+                "id",
+                opts.left_attr,
+                opts.right_attr,
+                opts.level_attr,
+                opts.tree_id_attr,
             )
-
-    def move_node(self, node, target, position="first-child"):
+        }
+        for node in nodes:
+            if node.id:
+                values = values_lookup[node.id]
+                for k, v in values.items():
+                    setattr(node, k, v)
+
+    def move_node(self, node, target, position="last-child"):
+        """
+        Vendored from mptt - by default mptt moves (with a save) then saves again
+        This is updated to just do the move, and no additional save.
```

Comment:
What are the benefits we get to vendoring this? Vendoring it means that changes in the underlying library could affect this code, so I don't think we should do it unless there's a significant benefit.

Reply:
Ah, sorry - looks like the code comment is outdated here. The save method does detection for changing parents (in case we carry out a move by setting the parent attribute on a model, or for insert operations), and will engage a tree lock as a result. I vendored this method in order to prevent engaging a second lock - but it's possible that this may be unnecessary.

In terms of this code changing - this method is a very thin wrapper around other MPTT methods, so the risk is relatively low from that perspective.

Reply:
Okay, I do agree it doesn't look like a particularly large change so I think in the interests of moving forward it's good to just leave this in, but I think it would be good to follow up on whether or not this is really needed. If not, then I think it's better to reduce the complexity and leave this to the mptt library, even if it's a little less efficient.